### PR TITLE
Update sonar-l10n-zh-10.5 for Sonarqube-10.5.

### PR DESCRIPTION
**Pull request by xuhuisheng**
Hi,
sonar-l10n-zh-plugin-10.5 has been released.

The main changes is support SonarQube 10.5.

The Download link and release notes: https://github.com/xuhuisheng/sonar-l10n-zh/releases/download/sonar-l10n-zh-plugin-10.5/sonar-l10n-zh-plugin-10.5.jar

The SonarCloud: https://sonarcloud.io/project/overview?id=xuhuisheng_sonar-l10n-zh

Please update the market place.

Thank you

Regards

### CODE REVIEW

File: l10nzh.properties [hunk 1]
```diff
@@ -2,16 +2,22 @@ category=Localization
...
-10.4.date=2024-02-10
-10.4.changelogUrl=https://github.com/xuhuisheng/sonar-l10n-zh/releases/tag/sonar-l10n-zh-plugin-10.4
-10.4.downloadUrl=https://github.com/xuhuisheng/sonar-l10n-zh/releases/download/sonar-l10n-zh-plugin-10.4/sonar-l10n-zh-plugin-10.4.jar
+10.5.description=Support SonarQube 10.5
+10.5.sqVersions=[10.4,10.LATEST]
+10.5.date=2024-04-13
+10.5.changelogUrl=https://github.com/xuhuisheng/sonar-l10n-zh/releases/tag/sonar-l10n-zh-plugin-10.5
+10.5.downloadUrl=https://github.com/xuhuisheng/sonar-l10n-zh/releases/download/sonar-l10n-zh-plugin-10.5/sonar-l10n-zh-plugin-10.5.jar
+
+10.4.1.description=Support SonarQube 10.4.1
```
Comment (ganncamp) [inline l10nzh.properties:16]: It's not clear to  me why you want to switch up this version number. There's no real need to exactly match your version number to the SQ version number, and by removing version 10.4 from this file, you leave the users on that version of your plugin high and dry. Please revert this version number to just 10.4

File: l10nzh.properties [hunk 2]
```diff
@@ -2,16 +2,22 @@ category=Localization
...
 defaults.mavenGroupId=org.codehaus.sonar-plugins.l10n
 defaults.mavenArtifactId=sonar-l10n-zh-plugin
 
-10.4.description=Support SonarQube 10.4
-10.4.sqVersions=[10.4,LATEST]
-10.4.date=2024-02-10
-10.4.changelogUrl=https://github.com/xuhuisheng/sonar-l10n-zh/releases/tag/sonar-l10n-zh-plugin-10.4
-10.4.downloadUrl=https://github.com/xuhuisheng/sonar-l10n-zh/releases/download/sonar-l10n-zh-plugin-10.4/sonar-l10n-zh-plugin-10.4.jar
+10.5.description=Support SonarQube 10.5
+10.5.sqVersions=[10.4,10.LATEST]
```
Comment (ganncamp) [inline l10nzh.properties:11]: Did you really mean to start this version compatibility at 10.4 and not 10.5?

You can only have one `publicVersion` marked as compatible with any given version of SonarQube.

**Review by ganncamp — CHANGES_REQUESTED** [review]

File: l10nzh.properties (hunk 2 above)
Comment (xuhuisheng) [reply l10nzh.properties:11]: I make a mistake on sqVersions range, I will fix it.

File: l10nzh.properties (hunk 1 above)
Comment (xuhuisheng) [reply l10nzh.properties]: I have reverted 10.4 part.

**Comment by xuhuisheng** [comment]
I squashed PR into one commit, please review. Thanks.

File: l10nzh.properties [hunk 3]
```diff
@@ -2,16 +2,22 @@ category=Localization
...
+10.5.downloadUrl=https://github.com/xuhuisheng/sonar-l10n-zh/releases/download/sonar-l10n-zh-plugin-10.5/sonar-l10n-zh-plugin-10.5.jar
+
 10.4.description=Support SonarQube 10.4
-10.4.sqVersions=[10.4,LATEST]
-10.4.date=2024-02-10
-10.4.changelogUrl=https://github.com/xuhuisheng/sonar-l10n-zh/releases/tag/sonar-l10n-zh-plugin-10.4
-10.4.downloadUrl=https://github.com/xuhuisheng/sonar-l10n-zh/releases/download/sonar-l10n-zh-plugin-10.4/sonar-l10n-zh-plugin-10.4.jar
+10.4.sqVersions=[10.4,10.4.*]
+10.4.date=2024-03-12
+10.4.changelogUrl=https://github.com/xuhuisheng/sonar-l10n-zh/releases/tag/sonar-l10n-zh-plugin-10.4.1
```
Comment (ganncamp) [inline l10nzh.properties:19]: Okay, I see now that you've released a new binary for 10.4. In all the other changes, I overlooked that yesterday.

Please 
* _Fully revert_ 10.4 to what it was 
* Except the compatibility range. That's currently what it needs to be.
* Put it in the `archivedVersions` list.
* And add a new version for your 10.4.1 binary that's compatible with `[10.4,10.4.*]` (just like 10.4).

File: l10nzh.properties [hunk 4]
```diff
@@ -1,18 +1,30 @@
...
 
+10.5.description=Support SonarQube 10.5
+10.5.sqVersions=[10.5,LATEST]
+10.5.date=2024-04-13
+10.5.changelogUrl=https://github.com/xuhuisheng/sonar-l10n-zh/releases/tag/sonar-l10n-zh-plugin-10.5
+10.5.downloadUrl=https://github.com/xuhuisheng/sonar-l10n-zh/releases/download/sonar-l10n-zh-plugin-10.5/sonar-l10n-zh-plugin-10.5.jar
+
 10.4.description=Support SonarQube 10.4
-10.4.sqVersions=[10.4,LATEST]
+10.4.sqVersions=10.4
```
Comment (ganncamp) [inline l10nzh.properties:17]: This should be `[10.4,10.4.1]` or `[10.4,10.4.*]`
If you don't leave this version marked compatible with 10.4.1, your users on that SQ version won't be prompted to upgrade to v10.4.1 of your plugin

**Review by ganncamp — CHANGES_REQUESTED** [review]

File: l10nzh.properties (hunk 4 above)
Comment (xuhuisheng) [reply l10nzh.properties]: I changed sqVersion to [10.4,10.4.1]

**Review by ganncamp — APPROVED** [review]

**Pull request merged**